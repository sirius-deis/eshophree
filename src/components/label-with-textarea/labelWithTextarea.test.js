import { fireEvent, render, screen } from "@testing-library/react";
import LabelWithTextarea from "./labelWithTextarea";

describe("LabelWithTextarea component", () => {
  it("should match snapshot", () => {
    const { container } = render(<LabelWithTextarea/>);
    expect(container).toMatchSnapshot();
  })
  it("should render label with provided text", () => {
    render(<LabelWithTextarea labelText="Textarea label *" withLabel/>);
    expect(screen.getByText(/textarea label/i)).toBeInTheDocument();
  })
  it("should not render label with provided text label if withLabel is false", () => {
    render(<LabelWithTextarea labelText="Textarea label" withLabel={false}/>);
    expect(screen.queryByText("Textarea label *")).toBeNull();
  })
  it("should render the textarea with correct attributes", () => {
    render(<LabelWithTextarea labelText="Textarea label" withLabel placeholder="Enter text" 
      name="test" 
      minLength={5} 
      maxLength={50} 
      value="Test"/>)
    const textarea = screen.getByPlaceholderText(/enter text/i);
    expect(textarea).toHaveAttribute("name", "test");
    expect(textarea).toHaveAttribute("minLength", "5");
    expect(textarea).toHaveAttribute("maxLength", "50");
    expect(textarea).toHaveValue("Test")
  })
  it("should handle textarea change event", () => {
    const setValueMock = jest.fn();
    render(<LabelWithTextarea setValue={setValueMock} placeholder="Enter text" value="Test" />);
    const textarea = screen.getByPlaceholderText(/enter text/i);
    fireEvent.change(textarea, { target: {value: 'New value' }});
    expect(setValueMock).toHaveBeenCalledWith("New value");
  });
})