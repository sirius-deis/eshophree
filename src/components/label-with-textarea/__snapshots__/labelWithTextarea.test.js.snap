// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelWithTextarea component should match snapshot 1`] = `
<div>
  <div>
    <textarea
      cols="70"
      maxlength="100"
      minlength="10"
      required=""
      rows="7"
    />
  </div>
</div>
`;
